### DISCUSSION TITLE
Cannot find module controllers

### DISCUSSION THREAD

**Discussion by allanlisboa** [adonisjs/core#3796] [Help]
My application cannot find the controllers generating an error. The error only occurs on the linux server.

{
  "message": "Cannot find module '/home/deploy/server/app/Controllers/Http/Auth/SigninsController'\nRequire stack:\n- /home/deploy/server/node_modules/@adonisjs/application/node_modules/@adonisjs/fold/build/src/Ioc/ImportAliases.js\n- /home/deploy/server/node_modules/@adonisjs/application/node_modules/@adonisjs/fold/build/src/Ioc/index.js\n- /home/deploy/server/node_modules/@adonisjs/application/node_modules/@adonisjs/fold/build/index.js\n- /home/deploy/server/node_modules/@adonisjs/application/build/index.js\n- /home/deploy/server/node_modules/@adonisjs/assembler/build/src/requireHook/ioc-transformer.js\n- /home/deploy/server/node_modules/@poppinss/utils/build/src/esmRequire.js\n- /home/deploy/server/node_modules/@poppinss/utils/build/index.js\n- /home/deploy/server/node_modules/@adonisjs/require-ts/build/src/Compiler/index.js\n- /home/deploy/server/node_modules/@adonisjs/require-ts/build/index.js\n- /home/deploy/server/node_modules/@adonisjs/assembler/build/src/requireHook/index.js\n- /home/deploy/server/node_modules/@adonisjs/assembler/build/register.js\n- internal/preload",
  "stack": "Error: Cannot find module '/home/deploy/server/app/Controllers/Http/Auth/SigninsController'\nRequire stack:\n- /home/deploy/server/node_modules/@adonisjs/application/node_modules/@adonisjs/fold/build/src/Ioc/ImportAliases.js\n- /home/deploy/server/node_modules/@adonisjs/application/node_modules/@adonisjs/fold/build/src/Ioc/index.js\n- /home/deploy/server/node_modules/@adonisjs/application/node_modules/@adonisjs/fold/build/index.js\n- /home/deploy/server/node_modules/@adonisjs/application/build/index.js\n- /home/deploy/server/node_modules/@adonisjs/assembler/build/src/requireHook/ioc-transformer.js\n- /home/deploy/server/node_modules/@poppinss/utils/build/src/esmRequire.js\n- /home/deploy/server/node_modules/@poppinss/utils/build/index.js\n- /home/deploy/server/node_modules/@adonisjs/require-ts/build/src/Compiler/index.js\n- /home/deploy/server/node_modules/@adonisjs/require-ts/build/index.js\n- /home/deploy/server/node_modules/@adonisjs/assembler/build/src/requireHook/index.js\n- /home/deploy/server/node_modules/@adonisjs/assembler/build/register.js\n- internal/preload\n    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:946:15)\n    at Function.Module._load (node:internal/modules/cjs/loader:787:27)\n    at Module.require (node:internal/modules/cjs/loader:1012:19)\n    at require (node:internal/modules/cjs/helpers:102:18)\n    at ImportAliases.resolve (/home/deploy/server/node_modules/@adonisjs/application/node_modules/@adonisjs/fold/build/src/Ioc/ImportAliases.js:95:23)\n    at ImportAliases.resolveAsync (/home/deploy/server/node_modules/@adonisjs/application/node_modules/@adonisjs/fold/build/src/Ioc/ImportAliases.js:113:25)\n    at Ioc.import (/home/deploy/server/node_modules/@adonisjs/application/node_modules/@adonisjs/fold/build/src/Ioc/index.js:280:42)\n    at Ioc.makeAsync (/home/deploy/server/node_modules/@adonisjs/application/node_modules/@adonisjs/fold/build/src/Ioc/index.js:391:40)\n    at IocResolver.call (/home/deploy/server/node_modules/@adonisjs/application/node_modules/@adonisjs/fold/build/src/Resolver/index.js:90:62)\n    at Object.PreCompiler.runRouteHandler [as fn] (/home/deploy/server/node_modules/@adonisjs/http-server/build/src/Server/PreCompiler/index.js:47:51)",
  "code": "MODULE_NOT_FOUND"
}

**Comment by tomgobich** [comment]
How are you defining the controller within your route definition?

It should look something like this:
```ts
Route.get('/signin', 'Auth/SigninsController.signin')

// -- OR --

Route.group(() => {
  Route.get('/signin', 'SigninsController.signin')
}).namespace('App/Controllers/Http/Auth')

**Comment by allanlisboa** [comment]
![Captura de Tela 2022-07-14 às 09 59 06](https://user-images.githubusercontent.com/37181884/178987960-00255718-a0ef-4c9b-b945-7945276c4a9a.png)

This is the route file

**Reply by allanlisboa** [reply]
The strange thing is that the server running on mac works perfectly

**Reply by tomgobich** [reply]
Does the controller file exist at the location it's trying to find it at, `/home/deploy/server/app/Controllers/Http/Auth/SigninsController`? Might be inside an additional build directory.
